test/framework/deployment_helpers.go

```diff
@@ -245,3 +283,124 @@ func WaitForDNSUpgrade(ctx context.Context, input WaitForDNSUpgradeInput, interv
 		return false, nil
 	}, intervals...).Should(BeTrue())
 }
+
+type DeployUnevictablePodInput struct {
+	WorkloadClusterProxy               ClusterProxy
+	ControlPlane                       *controlplanev1.KubeadmControlPlane
+	DeploymentName                     string
+	WaitForDeploymentAvailableInterval []interface{}
+}
+
+func DeployUnevictablePod(ctx context.Context, input DeployUnevictablePodInput) {
+	workloadClient := input.WorkloadClusterProxy.GetClientSet()
+	workloadDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      input.DeploymentName,
+			Namespace: "default",
```

Comment:
nit: I wonder if we can use the namespace that was created as part of the spec setup instead of "default" in order to avoid any cross test conflicts later down the road.